index.js

```diff
@@ -273,9 +273,12 @@ class ServerlessCustomDomain {
       certificateArn: givenCertificateArn,
     };
 
-    // This will return the distributionDomainName (used in changeResourceRecordSet)
-    const createDomain = this.apigateway.createDomainName(createDomainNameParams).promise();
-    return createDomain.then(data => data.distributionDomainName);
+    /* This will return the distributionDomainName (used in changeResourceRecordSet)
+      if the domain name already exists, the distribution domain name will be returned */
+    return this.getDomain().then(data => data.distributionDomainName).catch(() => {
```

Comment:
note that when getting regional domain the domain name will be in data.regionalDomainName